api/http/handler/stacks/create_compose_stack.go

```diff
@@ -49,8 +80,21 @@ func (handler *Handler) createComposeStackFromFileContent(w http.ResponseWriter,
 	if err != nil {
 		return &httperror.HandlerError{http.StatusInternalServerError, "Unable to check for name collision", err}
 	}
+
 	if !isUnique {
-		return stackExistsError(payload.Name)
+		stack, err := handler.DataStore.Stack().StackByName(payload.Name)
```

Comment:
Please keep in mind that `StackByName()` returns the first stack that matches that name, but there could be many stacks with the same name but on different environments. I think you need to do something like this to get the right one:

https://github.com/portainer/portainer/blob/develop/api/http/proxy/factory/kubernetes/namespaces.go#L55

Reply:
Have updated code according to your comments